src/components/Blog/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,153 @@
+import { useUserTracking } from '@/hooks/userTracking/useUserTracking';
+import type { StrapiMetaPagination } from '@/types/strapi';
+import ArrowBackIcon from '@mui/icons-material/ArrowBack';
+import ArrowForwardIcon from '@mui/icons-material/ArrowForward';
+import { darken, lighten, Typography, useTheme } from '@mui/material';
+import type { Dispatch, PropsWithChildren, SetStateAction } from 'react';
+
+import {
+  TrackingAction,
+  TrackingCategory,
+  TrackingEventParameter,
+} from '@/const/trackingKeys';
+import Link from 'next/link';
+import {
+  PaginationButton,
+  PaginationContainer,
+  PaginationIndexButton,
+} from './Pagination.style';
+
+interface PaginationProps {
+  isEmpty: boolean;
+  page: number;
+  setPage: Dispatch<SetStateAction<number>>;
+  pagination: StrapiMetaPagination;
+  categoryId: number | undefined;
+  id?: number | string;
+}
+
+export const Pagination = ({
+  page,
+  setPage,
+  pagination,
+  id,
+  categoryId,
+  isEmpty,
+}: PaginationProps) => {
+  const theme = useTheme();
+  const { trackEvent } = useUserTracking();
+  const handlePage = (page: number) => {
+    trackEvent({
+      category: TrackingCategory.BlogArticlesBoard,
+      label: 'click-pagination',
+      action: TrackingAction.ClickPagination,
+      data: {
+        [TrackingEventParameter.Pagination]: page,
+        [TrackingEventParameter.PaginationCat]: categoryId || '',
+      },
+    });
+    setPage(page);
+  };
+
+  const handleNext = () => {
+    if (page + 1 < pagination.pageCount) {
+      setPage((state) => state + 1);
+    } else {
+      setPage(0);
+    }
+    trackEvent({
+      category: TrackingCategory.BlogArticlesBoard,
+      label: 'click-pagination-next',
+      action: TrackingAction.ClickPagination,
+      data: {
+        [TrackingEventParameter.Pagination]: page,
+        [TrackingEventParameter.PaginationCat]: categoryId || '',
+      },
+    });
+  };
+
+  const handlePrev = () => {
+    if (page === 0) {
+      setPage(pagination.pageCount - 1);
+    } else {
+      setPage((state) => state - 1);
+    }
+    trackEvent({
+      category: TrackingCategory.BlogArticlesBoard,
+      label: 'click-pagination-prev',
+      action: TrackingAction.ClickPagination,
+      data: {
+        [TrackingEventParameter.Pagination]: page,
+        [TrackingEventParameter.PaginationCat]: categoryId || '',
+      },
+    });
+  };
+
+  return (
+    !isEmpty && (
```

Comment:
What this isEmpty does? the pagination component shouldn't be rendered if no data at the parent level